src/metrics/reef_indices.jl

```diff
@@ -0,0 +1,243 @@
+using FLoops, DataStructures
+
+"""
+    reef_condition_index(rc::AbstractArray, evenness::AbstractArray, sv::AbstractArray, juves::AbstractArray)::AbstractArray
+    reef_condition_index(rs)
+
+Estimates a Reef Condition Index (RCI) providing a single value that indicates the condition
+of a reef across four metrics: 
+- coral cover
+- evenness (coral diversity)
+- shelter volume, and
+- abundance of juveniles
+
+# Notes
+Juveniles are made relative to maximum observed juvenile density (51.8/m²)
+See notes for `juvenile_indicator()`
+
+# Arguments
+- `rc` : Relative coral cover across all groups
+- `evenness` : Evenness across all coral groups
+- `sv` : Shelter volume based on coral sizes and abundances
+- `juves` : Abundance of coral juveniles < 5 cm diameter
+
+# Returns
+NamedArray[timesteps ⋅ locations ⋅ scenarios]
+"""
+function _reef_condition_index(rc::AbstractArray, evenness::AbstractArray, sv::AbstractArray, juves::AbstractArray; threshold=2)::AbstractArray
+    # Compare outputs against reef condition criteria provided by experts
+    # These are median values for 8 experts.
+    condition_category = Symbol[:lower, :very_poor, :poor, :fair, :good, :very_good, :very_good]
+    criteria = NamedDimsArray([
+            0.0 0.0 0.0 0.0
+            0.05 0.15 0.175 0.15  # Very Poor [0 to these values]
+            0.15 0.25 0.3 0.25    # Poor
+            0.25 0.25 0.3 0.25    # Fair
+            0.35 0.35 0.35 0.25   # Good
+            0.45 0.45 0.45 0.35   # Very Good [these values and above]
+            Inf Inf Inf Inf       # Note: some metrics might return a value > 1.0
+        ],
+        condition=condition_category,
+        metric=[:RC, :E, :SV, :Juv]
+    )
+
+    index_metrics = zeros(size(rc)..., 4)
+    @floop for (idx, met) in enumerate([rc, evenness, sv, juves])
```

Comment:
Is @floop an extra efficient looping function?

Reply:
Provides multi-threading. Helps a little bit in terms of speed, but has to be used carefully.

Reply:
Thanks